telemetry/istio/src/main/scala/io/buoyant/telemetry/istio/MixerClient.scala

```diff
@@ -0,0 +1,111 @@
+package io.buoyant.telemetry.istio
+
+import com.google.local.DurationProto.Duration
+import com.twitter.finagle.buoyant.h2
+import com.twitter.finagle.Service
+import com.twitter.util.Future
+import istio.mixer.v1.{Attributes, Mixer, ReportRequest, ReportResponse, StringMap}
+
+private[telemetry] object MixerClient {
+
+  // TODO: build up this dictionary as requests are sent out,
+  // also consider using defaultWords, once per ReportRequest, rather than per Attributes
+  private[telemetry] val words = Seq[String](
+    // default words
+    "source.ip",
+    "source.port",
+    "source.name",
+    "source.uid",
+    "source.namespace",
+    "source.labels",
+    "source.user",
+    "target.ip",
+    "target.port",
+    "target.service",
+    "target.name",
+    "target.uid",
+    "target.namespace",
+    "target.labels",
+    "target.user",
+    "request.headers",
+    "request.id",
+    "request.path",
+    "request.host",
+    "request.method",
+    "request.reason",
+    "request.referer",
+    "request.scheme",
+    "request.size",
+    "request.time",
+    "request.useragent",
+    "response.headers",
+    "response.size",
+    "response.time",
+    "response.duration",
+    "response.code",
+
+    // custom words
+    "REQUEST_PATH", // index -32
+    "buoyant.svc.cluster.local",
+    "app",
+    "SOURCE_LABELS_APP",
+    "TARGET_LABELS_APP",
+    "version",
+    "TARGET_LABELS_VERSION"
+  )
+
+  private[telemetry] def mkReportRequest(): ReportRequest =
+    ReportRequest(
+      attributes = Seq[Attributes](
+        Attributes(
+          words = words,
```

Comment:
Since you're sharing the same word list for each of these `Attributes` messages, you can specify it once in the [`default_words`](https://github.com/istio/api/blob/master/mixer/v1/report.proto#L52) field of the `ReportRequest` and not have to repeat it per `Attributes` (even though there's only one `Attributes` right now).